client/src/components/atoms/TextFieldWithAction.tsx

```diff
@@ -15,36 +14,37 @@ interface TextFieldWithActionProps {
 const TextFieldWithAction: FunctionComponent<TextFieldWithActionProps> = (props: TextFieldWithActionProps) => {
     const [value, setValue] = useState("");
     const onTextChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        e.preventDefault();
         const newValue = e.target.value;
+        
+        props.onChange(newValue);
         setValue(newValue);
     }
     const handleKeyDown = (e: React.KeyboardEvent) => {
         if( e.key == 'Enter' ){
-            if (props.showAction){
-                if (value !== ""){
-                    props.onSubmit(value);
-                    setValue(""); 
-                }
+            e.preventDefault();
+            
+            if (value !== ""){
+                props.onSubmit(value);
+                setValue(""); 
             }
         }
     }
-    useEffect(() => {
-        props.onChange(value);
-    });
+    
     return (
     <div className="text-field-with-action">
         <div onKeyDown={handleKeyDown}>        
             <TextField input={value} isDisabled={false} isErroneous={props.isErroneous} onChange={onTextChange} name="text-field-action" placeholder={props.placeholder} type={props.type} iconClassName={props.iconClassName} ></TextField>
         </div>
-        {props.showAction && value!=="" && 
+        { value!=="" && 
             <div>
                 <a onClick={()=>{
                     if (value !== ""){props.onSubmit(value);}
                     setValue("");
                     }}>
                     <div className={"action" + (props.isErroneous ? " error" : "")}>
                         <div className="action-string">{props.actionString}</div>
-                        <div className="action-value"><span className="action-value-text">{value}</span></div>
+                        <div className="action-value"><span className="action-value-text">{value.substring(0,40)}</span></div>
```

Comment:
This isn't the input field. the div `action-value` shows a tag with the text with the input field
We're taking the substring to 40 chars for the tag bc otherwise the styling gets weird

![Untitled drawing](https://user-images.githubusercontent.com/20423590/114572219-4ccb4b00-9c45-11eb-9389-967fae7e477d.png)


sorry if my responses dont have any personality btw. lol not enough sleep

Reply:
My PR is also after an all-nighter... We'll get through this lol

Makes sense then!

Reply:
<<<<<<333333333